src/Applications/GEOSdas_App/ldas_add2rc

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+MYNAME="ldas_add2rc"
+if [[ $# -ne 2 ]]; then 
+   echo " "
+   echo " NAME "
+   echo " "
+   echo "  $MYNAME  - reconcile GEOSldas and AGCM configurations in weakly-coupled LADAS "
+   echo " "
+   echo " SYNOPSIS  "
+   echo " "
+   echo "  $MYNAME  FVHOME  LDHOME"
+   echo " "
+   echo " where"
+   echo "   FVHOME  -  location of  ADAS experiment "
+   echo "   LDHOME  -  location of  LDAS experiment "
+   echo " "
+   echo " DESCRIPTION"
+   echo " "
+   echo "    This procedure inserts select GEOSldas resource parameters into AGCM.rc and " 
+   echo "    verifies consistency between LDAS.rc and AGCM.rc for other resource parameters "
+   echo " "
+   echo "   $MYNAME should be run after fvsetup and ldas_setup  "
+   exit 0
+fi
+
+FVHOME="$1" 
+LDHOME="$2" 
+
+# ----------------------------------------------------------------
+# extract the following parameters from LDAS.rc and add to AGCM.rc
+
+cd $FVHOME/run
+cp AGCM.rc.tmpl  AGCM.rc.tmpl.original 
+
+list="LANDASSIM_DT LANDASSIM_T0 "
+  for var in $list
+do
+    grep $var $LDHOME/run/LDAS.rc >> add.txt
+done
+
+sed -i '/LDAS_INCR/r add.txt' AGCM.rc.tmpl
+
+echo " The following land parameters are added to AGCM.rc: " 
+for var in $list
+do
+echo "$(grep $var add.txt)"
+done
+
+# add same resource parameters into AGCM.rc for atm ens
+mv add.txt $FVHOME/run/atmens/. 
+cd $FVHOME/run/atmens/ 
+cp AGCM.rc.tmpl AGCM.rc.tmpl.original 
+
+sed -i '/LDAS_INCR/r add.txt' AGCM.rc.tmpl
+
+rm -rf add.txt
+
+# ------------------------------------------------------------------------
+# check consistency of the following parameter between LDAS.rc and AGCM.rc
+
+clist=" Z0_FORMULATION:.*  LAND_PARAMS:.*  LSM_CHOICE:.* "
+  for var in $clist 
+do 
+Lset="$(grep $var $LDHOME/run/LDAS.rc)"
+Aset="$(grep $var $FVHOME/run/AGCM.rc.tmpl)"
+lv="$(echo $Lset | cut -d " " -f 2)"
+av="$(echo $Aset | cut -d " " -f 2)"
+if [ "$lv" != "$av" ]; then
+echo " " 
+echo "Please repeat setup such that the following parameter values are consistent in AGCM.rc and LDAS.rc:"
+echo "in AGCM.rc: $Aset "  
+echo "in LDAS.rc: $Lset "
+echo " " 
```

Comment:
This script should halt the LADAS setup if the LDAS.rc and ADAS.rc have inconsistent values.  I don't see this "exit" or "stop" condition, but perhaps I'm missing something.

Reply:
This script will be run after fvsetup and ldas_setup procedures are done. Both adas exp and ldas exp dir are setup with LDAS.rc and AGCM.rc.    If the script finds inconsistent values,  all it can do  is to print warning message. We wanted to check all parameters in the list ( not stop and exit at the first found inconsistency),  then it reaches  the end,

Reply:
This script is run after fvsetup and ldas-setup ( so that AGCM.rc and LDAS.rc are in place in exp/run ) .  It should go through the given list of parameters and prints out warning message if any one is inconsistent,  and not exit till reaching the end of the parameter list.

Reply:
It makes sense to run through all parameter values and tabulate the ones that are inconsistent before exiting.
But what still seems to be missing is a "return code" and controlled exit.  This script should return something to the main script that tells the main script to exit/stop with a message.  If the list of inconsistent parameters is written by this script, then the main script can simply exit with a brief error message, something like "ldas_add2rc: inconsistent ADAS and LDAS config"

Reply:
This script is not called by any main script.  It is a standalone "post setup check"  after fvsetup and ldas_setup are finished.

Reply:
Thanks, I see.  I'm still not sure how best to sort out this chicken-and-egg scenario.  Could we possibly have fvsetup call ldas_setup?  
If I understand the discussion correctly, fvsetup is provided with LDROOT, so it should be able to run ldas_setup.  One challenge is that ldas_setup requires the LDAS "exeinp" and "batinp" config files, which would then also have to be provided to fvsetup.  But the user has to have these files anyway, so providing them to fvsetup doesn't really change anything.  
Another complication is that g5_modules is probably different between the ADAS and the LDAS.  If ldas_setup is called at the very end of fvsetup, could we have fvsetup source (LDAS) g5_modules just before calling ldas_setup? 
If that's possible, then fvsetup could also call ldas_add2rc after calling ldas_setup. 
Alternatively, maybe a new main script, fvsetup_ladas, could be added to the ADAS repository.  This script would first find out LDROOT, LDHOME, LDAS "exeinp"/"batinp", then call fvsetup, then call ldas_setup, and finally ldas_add2rc.
Maybe I'm too optimistic about constructing a single master script that sets up both the ADAS and the LDAS, but if the LADAS is to be used more broadly we have to make the setup as easy and as robust as possible.
For now, we can wait until Ricardo has had a chance to take a closer look.  I imagine he will have some thoughts on this as well.